### DISCUSSION TITLE
Enable automatic branch deletion for repository with api

### DISCUSSION THREAD

**Discussion by ojintoad** [community/community#24530] [Apps, API and Webhooks]
<p><a href="https://help.github.com/en/articles/managing-the-automatic-deletion-of-branches" rel="nofollow noopener">https://help.github.com/en/articles/managing-the-automatic-deletion-of-branches</a></p>
<p>This exists, which is nice</p>
<p><a href="https://developer.github.com/v3/repos/" rel="nofollow noopener">https://developer.github.com/v3/repos/</a></p>
<p>This exists, which is also nice</p>
<p>Unfortunately, the latter doesn’t seem to acknowledge the former, and AFAICT there is no means of managing the automtic deletion of branches through the API, only the UI.</p>
<p>Did I miss where it’s actually located within the APIs? If I didn’t and it can’t be managed through APIs, is this on the roadmap to add? Excited to do this for the many repos we manage.</p>
<p>Thank you!</p>

**Comment by ojintoad** [comment]
<p>FWIW as an enterprise customer I did contact support and this was the response I got:</p>
<blockquote>
<p>Hi Todd, Thank you for sending this in. We have an open feature request for this and have added your feedback. I cannot say if or when this will be implemented. Keep an eye on the release notes for information on when and if this makes it into the product. ​Additionally, you can keep an eye on api changes to API v3 and GraphQL API v4 here, respectively: ​ - <a href="https://developer.github.com/changes/" rel="nofollow noopener">https://developer.github.com/changes/</a> - <a href="https://developer.github.com/v4/changelog/" rel="nofollow noopener">https://developer.github.com/v4/changelog/</a></p>
</blockquote>
<p>So hopefully it will get implemented and added. If you’re an enterprise/professional customer reading this, you should probably contact support to add your voice towards getting the feature added!</p>

**Comment by MChevy422** [comment]
<p>Hello <span><a class="mention" href="/ojintoad">@ojintoad</a></span>, </p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information. As you’ll notice, your Topic is now in the <strong>Github Action</strong> board. No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Also, thank you for your feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feedback in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.<br><br>Once again, thank you for your input!<br></p>
<p>-MChevy422</p>

**Comment by ojintoad** [comment]
<p>Respectfully, I’m confused. Nothing about managing this has to do with github actions and it’s actually just a setting on the repo that I’m asking to be made available on the API. Why would this be moved to GitHub Actions?</p>

**Comment by ojintoad** [comment]
<p>Hi <a class="mention" href="/mchevy422">@mchevy422</a> - I responded to this awhile ago flagging I think this move is incorrect given it is about API calls. Can you respond to my prior inquiry please?</p>

**Comment by MChevy422** [comment]
<p><a class="mention" href="/ojintoad">@ojintoad</a></p>
<p>I apologize, it was an honest mistake. I went ahead and moved it back to the correct location.</p>
<p>MChevy422</p>

**Comment by ashirley** [comment, marked as answer]
<p>Note that <code>delete_branch_on_merge</code> is now supported (<a href="https://developer.github.com/v3/repos/#update-a-repository" rel="nofollow noopener">https://developer.github.com/v3/repos/#update-a-repository</a>)</p>
